template/app/src/main/java/com/xmartlabs/template/helper/GeneralCompletableSubscriber.java

```diff
@@ -0,0 +1,51 @@
+package com.xmartlabs.template.helper;
+
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.StringRes;
+
+import com.xmartlabs.template.ui.common.TemplateView;
+
+import java.lang.ref.WeakReference;
+
+import io.reactivex.CompletableObserver;
+import io.reactivex.disposables.Disposable;
+
+public class GeneralCompletableSubscriber implements CompletableObserver {
+  @NonNull
+  private WeakReference<TemplateView> viewReference;
+
+  public GeneralCompletableSubscriber() {
+    this(null);
+  }
+
+  public GeneralCompletableSubscriber(@Nullable TemplateView templateView) {
+    viewReference = new WeakReference<>(templateView);
+  }
+
+  @Override
+  public void onSubscribe(@NonNull Disposable disposable) {}
+
+  @Override
+  public void onComplete() {}
+
+  @Override
+  public void onError(@NonNull Throwable throwable) {
+    TemplateView view = viewReference.get();
+    if (alertOnError(throwable) && view != null && view.isViewAlive()) {
+      view.showError(throwable, getErrorMessage(throwable));
+    }
+  }
+
+  @Nullable
+  @StringRes
+  protected Integer getErrorMessage(Throwable throwable) {
```

Comment:
It's uglier to use, you'll have to use Optional.of(R.string.bla) instead of just R.string.bla.
I'd prefer the caller to wrap it if needed